test/built-ins/DataView/prototype/setFloat32/index-check-before-value-conversion.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2015 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es7id: 24.2.1.2
+description: >
+  Throws a RangeError if the index is negative or non-integral number.
+info: >
+  SetViewValue ( view, requestIndex, isLittleEndian, type, value )
+
+  ...
+  3. Let numberIndex be ToNumber(requestIndex).
+  4. Let getIndex be ? ToInteger(numberIndex).
+  5. If numberIndex ≠ getIndex or getIndex < 0, throw a RangeError exception.
```

Comment:
@anba, what do you think about adding a test case with an object coerced to an invalid number?